apps/cli/src/exec/genext2fs.ts

```diff
@@ -0,0 +1,68 @@
+import { parse, SemVer } from "semver";
+import { execaDockerFallback } from "./util";
+
+export const fromDirectory = async (options: {
+    cwd?: string;
+    extraSize: number;
+    image?: string;
+    input: string;
+    output: string;
+}) => {
+    const { cwd, extraSize, image, input, output } = options;
+    const blockSize = 4096; // fixed at 4k
+    const extraBlocks = Math.ceil(extraSize / blockSize);
+    const args = [
+        "--block-size",
+        blockSize.toString(),
+        "--faketime",
+        "--root",
+        input,
+        "--readjustment",
+        `+${extraBlocks}`,
+        output,
+    ];
+    return execaDockerFallback("xgenext2fs", args, { cwd, image });
+};
+
+export const fromTar = async (options: {
+    cwd?: string;
+    extraSize: number;
+    image?: string;
+    input: string;
+    output: string;
+}) => {
+    const { cwd, extraSize, image, input, output } = options;
+    const blockSize = 4096; // fixed at 4k
+    const extraBlocks = Math.ceil(extraSize / blockSize);
+
+    const args = [
+        "--block-size",
+        blockSize.toString(),
+        "--faketime",
+        "--tarball",
+        input,
+        "--readjustment",
+        `+${extraBlocks}`,
+        output,
```

Comment:
This is very similar to the arguments from `fromDirectory`, maybe we could reduce this duplication and jave a `baseArgs` somewhere  that could be just appended with what is particular to `fromTar` and `fromDirectory`.

Reply:
done